src/packages/@app/pages/RestExplorer/RestExplorerPage.ViewModel.ts

```diff
@@ -1883,6 +1894,96 @@ class RestExplorerViewModel
     return response;
   };
 
+  /**
+   * Generates stream wise an AI response based on the given prompt.
+   *
+   * @param prompt - The prompt to send to the AI assistant service.
+   * @returns A promise that resolves to the response from the AI assistant service.
+   */
+  public generateStreamAiResponse = async (prompt = "") => {
+    // Set the request state to indicate that a response is being generated
+    await this.updateRequestState({ isChatbotGeneratingResponse: true });
+    let componentData = this._tab.getValue();
+    const apiData = {
+      body: componentData.property.request.body,
+      headers: componentData.property.request.headers,
+      method: componentData.property.request.method,
+      queryParams: componentData.property.request.queryParams,
+      url: componentData.property.request.url,
+      auth: componentData.property.request.auth,
+    };
+    const socketValue: Socket =
+      await this.aiAssistentWebSocketService.sendPromptMessage({
+        text: prompt,
+        instructions: `You are an AI Assistant, responsible for answering API related queries. Give the response only in markdown format. Only answer questions related to the provided API data and API Management. Give to the point and concise responses, only give explanations when they are asked for. Always follow best practices for REST API and answer accordingly. Utilize the provided api data ${JSON.stringify(
```

Comment:
lets push this instructions to backend for flexibility, previous implementation isn't that correct